pkg/controller/manager/clusternetwork/controller.go

```diff
@@ -144,10 +144,16 @@ func (h Handler) setNadReadyLabel(cn *networkv1.ClusterNetwork) error {
 	}
 
 	for _, nad := range nads {
+		if nad.DeletionTimestamp != nil {
+			continue
+		}
 		if nad.Labels[utils.KeyNetworkReady] == isReady {
 			continue
 		}
 		nadCopy := nad.DeepCopy()
+		if nadCopy.Labels == nil {
+			nadCopy.Labels = make(map[string]string)
+		}
```

Comment:
should we move this after the check for `nad.DeletionTimestamp` since we are checking 
```
if nad.Labels[utils.KeyNetworkReady] == isReady {
			continue
		}
```

if nad.Labels are not going to be empty, then nadCopy is going to inherit the labels too.

```suggestion
		if nad.Labels == nil {
			nad.Labels = make(map[string]string)
		}
```